spec/models/accounting/quickbooks/connector_spec.rb

```diff
@@ -0,0 +1,99 @@
+require 'rails_helper'
+
+RSpec.describe Accounting::Quickbooks::Connector, type: :model do
+  let(:valid_token) { 'lvprdxMSsckHORgjp9RCmVaF6anST6VWIVU84eQempNRZy0f' }
+  let(:invalid_token) { '' }
+  let(:valid_secret) { '295VilDqvyRaUTMFOIzYjUL1eFJCApCwBRItPeWf' }
+  let(:invalid_secret) { '' }
+  let(:valid_realm_id) { '193514472376479' }
+  let(:invalid_realm_id) { '' }
+
+  shared_examples_for 'no token is present' do
+    it 'should not be connected' do
+      expect(connector.connected?).to be_falsey
+    end
+    it 'should not be renewable' do
+      expect(connector.renewable?).to be_falsey
+    end
+    it 'should not be expired' do
+      expect(connector.expired?).to be_falsey
+    end
+  end
+
+  context 'with new token' do
+    subject(:connector) do
+      Accounting::Quickbooks::Connector.new(quickbooks: { token: valid_token, secret: valid_secret, realm_id: valid_realm_id, token_expires_at: 180.days.from_now.utc })
```

Comment:
The fact that `quickbooks:` needs to be passed here should be an immediate code smell. Do you see why?